### PR TITLE
Rework scanner tests

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

**Comment by sschuberth** [comment]
If you review commit-by-commit, please note that GitHub messed with the order of commits (it shows them sorted by authoring time, not commit time).

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt [hunk 1]
```diff
@@ -34,35 +37,71 @@ import java.util.TreeSet
...
 
             resultsFile.isFile shouldBe true
             result.summary.fileCount shouldBe 1
             result.summary.licenses shouldBe expectedFileLicenses
-            result.summary.licenseFindings.all {
-                it.location.path == licenseFilePath.invariantSeparatorsPath
-            } shouldBe true
+
+            // Check for relative paths in the result.
+            result.summary.licenseFindings.forEach {
```
Comment (fviernau) [inline scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: How about re-writing the assertion in a way which makes the code comment unnecessary? Like e.g.

`result.summary.licenseFindings.map { it.location.path} shouldContainExactlyInAnyOrder listOf(
   "relpath/LICENSE", 
     ...  
)`

File: scanner/src/main/kotlin/LocalScanner.kt [hunk 2]
```diff
@@ -434,16 +434,14 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
      * [scanned path][scanPath].
      */
     protected fun relativizePath(scanPath: File, scannedFilename: File): String {
-        val relativePathToScannedFile = if (scanPath.isFile) {
-            if (scanPath.isAbsolute) {
-                // Scanners are called with the current working directory unchanged. To avoid absolute paths, make a
-                // path to a file relative to the current working directory in the absence of a scan root directory.
-                scanPath.relativeTo(File(".").absoluteFile)
+        val relativePathToScannedFile = if (scannedFilename.isAbsolute) {
```
Comment (fviernau) [inline scanner/src/main/kotlin/LocalScanner.kt]: commit-msg: cannot see the added comments, I assume they went into a different commit meanwhile?

Comment (sschuberth) [reply scanner/src/main/kotlin/LocalScanner.kt]: They are in https://github.com/heremaps/oss-review-toolkit/pull/2327/commits/918ec04558596bdeb0091be0308458385eb57922 (but that SHA1 will change with my next force-push).

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt [hunk 3]
```diff
@@ -34,35 +37,71 @@ import java.util.TreeSet
...
+            val result = scanner.scanPath(inputDir.resolve("LICENSE"), resultsFile)
 
             resultsFile.isFile shouldBe true
             result.summary.fileCount shouldBe 1
             result.summary.licenses shouldBe expectedFileLicenses
-            result.summary.licenseFindings.all {
-                it.location.path == licenseFilePath.invariantSeparatorsPath
-            } shouldBe true
+
+            // Check for relative paths in the result.
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: So far our (unofficial) convention was to have blank lines before blocks / before starting a comment (that refers to following lines), and I'd prefer to stick to that style.

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt [hunk 4]
```diff
@@ -34,35 +37,71 @@ import java.util.TreeSet
...
+            result.summary.licenseFindings.forEach {
+                File(it.location.path) shouldNotStartWithPath inputDir
+            }
+        }
+
+        "Scanning a directory succeeds".config(tags = testTags) {
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: No, but I don't think they need to be, as test cases of the same spec are never executed in parallel.

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt [hunk 5]
```diff
@@ -34,35 +37,71 @@ import java.util.TreeSet
 
 abstract class AbstractScannerTest : StringSpec() {
     protected val config = ScannerConfiguration()
-    private val licenseFilePath = File("../LICENSE")
+    private val commonlyDetectedFiles = listOf("LICENSE", "LICENCE", "COPYING")
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: I'm not getting the rationale, could you please try to rephrase?

I'm using "commonly" synonymous to "usually" here, i.e. I want to express that these are file names that are usually recognized by scanners. The thinking comes from the Licensee scanner, which only looks at files that match certain patterns.

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt [hunk 6]
```diff
@@ -34,35 +37,71 @@ import java.util.TreeSet
...
     abstract val testTags: Set<Tag>
 
     override fun beforeSpec(spec: Spec) {
         super.beforeSpec(spec)
+        inputDir = createTempDir(ORT_NAME, javaClass.simpleName)
+
+        // Copy our own root license under different names to a temporary directory so we have something to operate on.
+        val ortLicense = File("../LICENSE")
+
+        // This is loosely based on the patterns from
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: See above, they are relevant to the Licensee scanner. If I named a file `FOOBAR` then *Licensee* would not scan it even if it is in the directory to scan.

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt (hunk 1 above)
Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: Without the comment, I find it to be less clear what property of the string the assertion is checking for. For the single file the string would be just `"LICENSE"` which does not really tell the reader that this should be a relative path.

But maybe I'll still play with the `listOf()` idea as it also asserts for the number of license findings, which is nice.

Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: After looking at the code again, I've simply removed the comment and blank line as `shouldNotStartWithPath` indeed makes it clear that paths should be relative.

File: scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: I assumed that the Scanner class did not distinguish based on the file name and I was wrong as I did not know the Licensee details.

[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/funTest/kotlin/scanners/AbstractScannerTest.kt]: For completeness: It's not (our) scanner class that makes the distinction based on files names, but Licensee itself.

**Comment by sschuberth** [comment]
Licensee test fails because of https://github.com/libgit2/rugged/issues/791.

**Pull request merged**